biohansel/create/cluster_generator.py

```diff
@@ -0,0 +1,236 @@
+from typing import Dict, Set, List
+
+import numpy as np
+import pandas as pd
+import scipy as sp
+
+
+from collections import defaultdict
+from scipy.cluster.hierarchy import fcluster, linkage
+from scipy.spatial.distance import pdist
+
+
+def find_clusters(df: pd.DataFrame, min_group_size: int) -> Dict[str, int]:
```

Comment:
What about user specified distance threshold levels? If `None` then you can use the unique distances from the distance matrix. This function should be returning all the intermediate outputs as well. Please look into creating an [attrs](http://www.attrs.org/en/stable/) class to store this information like the `Subtype` class.

Reply:
yes this has been addressed in 94b2c0213878ff804e065eb970dc3023030acfcc